core/primitives-core/src/convert.rs

```diff
@@ -0,0 +1,24 @@
+use crate::types::Gas;
+
+pub trait ToGas<T> {
+    fn tera(_: T) -> Self;
+}
+
+impl ToGas<f64> for Gas {
+    fn tera(tera_gas: f64) -> Self {
+        (tera_gas * f64::powf(10_f64, 12_f64)) as Gas
```

Comment:
floats are definitely not the way to go

Reply:
I think this should be done in near-sdk-rs
if not float what should be used, my first version was made for a string, but @matklad  [showed that it is better to use float](https://github.com/near/near-sdk-rs/pull/368#discussion_r617464033), I was confused